cluster/store/service.go

```diff
@@ -330,12 +335,14 @@ func (s *Service) Query(ctx context.Context, req *cmd.QueryRequest) (*cmd.QueryR
 	if s.store.IsLeader() {
 		return s.store.Query(req)
 	}
+	s.store.WaitForLeader(ctx, time.Duration(6*time.Second))
```

Comment:
thoughts(non-blocking): should we make this a parameter in our config ? Something like `raft_leader_request_timeout` ?

Reply:
Also I think we have similar code in `Apply` where we fetch the leader and proxy to it, should it have the same change applied ?

Reply:
I had the same thought, but thought could introduce it later in different PR to not have huge change on this PR